src/main/java/Bambot.java

```diff
@@ -0,0 +1,179 @@
+        case "bye":
+            printByeMessage();
+            return true;
+        case "todo":
+            handleToDoCommand(input);
+            break;
+        case "deadline":
+            handleDeadlineCommand(input);
+            break;
+        case "event":
+            try {
+                handleEventCommand(input);
+            } catch (BambotException e) {
+                System.out.println(e.getMessage());
+            }
+            break;
+        case "delete":
+            handleDeleteCommand(input);
+            break;
+        case "list":
+            printList();
+            break;
+        case "mark":
+            handleMarkCommand(input);
+            break;
+        case "unmark":
+            handleUnmarkCommand(input);
+            break;
+        default:
+            System.out.println(message + " is not a valid command");
+            System.out.println(DIVIDER);
+            break;
+        }
+        return false;
+    }
+
+    private static void handleUnmarkCommand(String input) {
+        int unmarkIndex = -1;
+        try {
+            unmarkIndex = Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            System.out.println("Input must be a number (Example: mark 2)");
+        }
+        try {
+            myList.get(unmarkIndex - 1).unmarkTask();
+            printList();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Index is out of bounds:" + input);
+            System.out.println("Pls use a number in the list");
+        }
+    }
+
+    private static void handleMarkCommand(String input) {
+        int markIndex = -1;
+        try {
+            markIndex = Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            System.out.println("Input must be a number (Example: mark 2)");
+        }
+        try {
+            myList.get(markIndex - 1).markTask();
+            printList();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Index is out of bounds:" + input);
+            System.out.println("Pls use a number in the list");
+            printList();
+        }
+    }
+
+    private static void handleDeleteCommand(String input) {
+        int removeIndex = -1;
+        try {
+            removeIndex = Integer.parseInt(input) - 1;
+        } catch (NumberFormatException e) {
+            System.out.println("Input must be a number (Example: remove 2)");
+        }
+        try {
+            ListItem removedTasked = myList.remove(removeIndex);
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(" " + removedTasked);
+            System.out.println("Now you have " + myList.size() + " tasks in the list");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Index is out of bounds:" + input);
+            System.out.println("Pls use a number in the list");
+            printList();
+        }
+    }
```

Comment:
Could consider condensing the try catch block as shown below to make the code more concise:
```suggestion
try {
// logic
} catch (NumberFormatException e) {
// handle NumberFormatException
} catch (IndexOutOfBoundsException e) {
// handle IndexOutOfBoundsException 
}